include/oneapi/dpl/pstl/ranges/nanorange.hpp

```diff
@@ -8006,16 +8006,22 @@ class counted_iterator
     template <typename II = I, ::std::enable_if_t<input_iterator<II>, int> = 0>
 #                    endif
     friend constexpr iter_rvalue_reference_t<I>
-    iter_move(const counted_iterator& i) noexcept(noexcept(ranges::iter_move(i.current_)))
+#                   ifndef _MSC_VER
+    // GCC 15 and later cannot find i.current_ outside of immediate context
+    iter_move(const counted_iterator<II>& i)
+#                   else
+    iter_move(const counted_iterator& i)
+#                   endif
+        noexcept(noexcept(ranges::iter_move(i.current_)))
```

Comment:
This is super confusing to read.  
I would recommend redundantly providing the return type and combining the two `#ifndef` blocks.

Reply:
Done